### DISCUSSION TITLE
Bootstrap, direct DOM manipulation & Vue's virtual DOM #310

### DISCUSSION THREAD

**Discussion by TechAkayy** [twbs/bootstrap#36011] [Q&A] [answered]
Hi there, 

This topic is w.r.t using Bootstrap (not just CSS, but JS components) in general with virtual-dom based frameworks like Vue & React. 

I would like to ask a very fundamental question, surprisingly I haven't really seen this discussed anywhere online expect for this stackoverflow [comment](https://stackoverflow.com/questions/65547199/using-bootstrap-5-with-vue-3#comment125680515_66386443)

![image](https://user-images.githubusercontent.com/73772701/157210617-f40dae7e-d044-4769-923b-743e88fec64e.png)

I know mixing Vue (virtual dom) + Jquery/Vanilla JS (that does direct dom manipulator) is an [anti-pattern](https://forum.vuejs.org/t/how-vue-2-0-work-with-jquery/3591/3), [anti-pattern](https://forum.vuejs.org/t/when-is-it-ok-to-directly-manipulate-the-dom/45818/3). 

Can anyone advise how you are using bootstrap JS components with Vue/React. If not, can we assume & establish that only the CSS part of Bootstrap is safe to use with Vue/React? 

I was also wondering about wrapper libraries like https://github.com/bootstrap-vue/bootstrap-vue/discussions/6931 & [bootstrap-vue-3](https://github.com/cdmoro/bootstrap-vue-3/discussions/310), if by design, they got some smarts to avoid direct dom manipulation by bootstrap.min.js. I have posted this same topic in their respective repos, but thought I will ask the question for any users here familiar with these wrapper libraries. 

Sorry, if it turns out to be a silly question, it's quite possible I might have misunderstood somethings! Thanks in advance.

**Comment by ascensus-mdb** [comment]
There are dedicated solutions for this. Have a look at:
https://mdbootstrap.com/docs/b5/vue/
https://mdbootstrap.com/docs/b5/react/

**Reply by mariusa** [reply]
How does that work with DOM manipulation? Does it use only CSS from bootstrap, no JS?

**Comment by gerardnico** [comment, marked as answer]
In other js framework library such as react, Vue, you reuse only the stylesheet. The js component are normally rewritten in a react/Vue library 

See Bootstrap react, 
https://react-bootstrap.github.io/

reactstrap,
https://reactstrap.github.io/

And more
